### ISSUE TITLE
On Login Process, "Open in Facebook" popup causes failure of the login.

### ISSUE THREAD

**Issue by hdemircioglu** [facebook/facebook-sdk-for-unity#545]
## Checklist

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-sdk-for-unity/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Unity Editor Version: `2019.4.23f1`
- Unity SDK Version: `9.1.0`
- Installation Platform & Verison: `[iOS|Android|WebGL]` version `14.4.2`

## Goals

What do you want to achieve?
Trying to login with read permissions of public_profile, email, user_friends.

## Expected Results

What do you expect to happen?
After clicking open in Facebook, and giving allowance to the app popup needs to close with success result.

## Actual Results

What actually happened? Can you provide a stack trace?
After clicking open in Facebook, it launches the app. When I click continue on allowance of the app to login, it turns back to the same page. So, I left with the only option to cancel.

## Steps to Reproduce

What are the steps necessary to reproduce this issue?
1- Click Login from the app
2- Select Open in "Open in Facebook popup"
3- Allow app to login by selecting continue, from "Previously logged in to the game. Would you like to continue?"

## Code Samples & Details

Please provide a code sample, as well as any additional details, to help us track down the issue. If you can provide a link to a test project that allows us to reproduce the issue, this helps us immensely in both the speed and quality of the fix.

![IMG_0002](https://user-images.githubusercontent.com/352879/115229755-18abca80-a11c-11eb-8df7-4f70fb12b2c9.PNG)
![IMG_0005](https://user-images.githubusercontent.com/352879/115229974-5d376600-a11c-11eb-86ba-2232f94427a3.PNG)
![IMG_0003](https://user-images.githubusercontent.com/352879/115230088-8b1caa80-a11c-11eb-8bce-20e20396d8fd.PNG)

**Comment by johnny-tictoc** [comment]
This is also happening to us, but on iOS only. We are using v8.0 of the SDK. The SDK seems to keep passing us between Facebook App and the WebView. Uninstalling the Facebook app allows us to successfully login.

![step1](https://user-images.githubusercontent.com/18489868/115382019-c4314980-a206-11eb-804f-5634e3830a45.png)
![step2](https://user-images.githubusercontent.com/18489868/115382018-c398b300-a206-11eb-84a2-7bb496b2d64b.png)
![step3](https://user-images.githubusercontent.com/18489868/115382005-c09dc280-a206-11eb-92ae-7d65ba3cd554.png)

**Comment by legoritma** [comment]
Is your app have AppsFlyer?  Your case reminds me [#519](https://github.com/facebook/facebook-sdk-for-unity/issues/519)

**Comment by johnny-tictoc** [comment]
@legoritma Yes, we are using Appsflyer. I'll try out the fix in that link. Thanks!

**Comment by hdemircioglu** [comment]
Hi, 
Thanks for the share. I tried AppsFlyerShouldSwizzle solution. Now, I have that line in the Xcode;
![image](https://user-images.githubusercontent.com/352879/116058738-ce30cd80-a688-11eb-8dc2-cdc93748c6b3.png)

But, the problem still continues. It does not solve it.

**Comment by ppansy** [comment]
Hi @hdemircioglu, thanks for reporting. A couple questions to helper us understand your issue:
- Which platform(s) are you seeing this issue? The screenshot looks like iOS but just confirming
- Can you share any logs & code samples?
- Would you be ok with sharing if you are using any other third party libraries other than Appsflyer?
- If possible, please provide a link to a test project, this would speed up our fix tremendously

**Comment by hdemircioglu** [comment]
Hi @ppansy,

Answers to the questions:
- Only on IOS, and when the Facebook app is installed.
- Log is as follows; 
[FacebookLogin.log](https://github.com/facebook/facebook-sdk-for-unity/files/6397378/FacebookLogin.log)
- Third-party apps: AppsFlyer, Firebase, Flurry, GameAnalytics, Ironsource
- Test build: https://testflight.apple.com/join/CTkRln2t

Thanks for the help.

**Comment by ppansy** [comment]
This seems similar to #546 
Can your try out the following fix?
https://github.com/facebook/facebook-sdk-for-unity/issues/546#issuecomment-832427152

**Comment by hdemircioglu** [comment]
![image](https://user-images.githubusercontent.com/352879/117282630-a0f4d400-ae6d-11eb-9b3c-fe620259dfcf.png)

When I searched the term, I end up with the result in the image. I also double-checked Swizzling condition by logging it in the initialization with the code below and it is logged as "true". My problem looks like not related to the AppDelegateListener. Am I missing something? 

```
        id swizzleFlag = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppsFlyerShouldSwizzle"];
        BOOL shouldSwizzle = swizzleFlag ? [swizzleFlag boolValue] : NO;
        
        NSLog(@"Swizzle: %s", shouldSwizzle ? "true" : "false");
        
        if(!shouldSwizzle){
            UnityRegisterAppDelegateListener(self);
        }
```

**Comment by ppansy** [comment]
From the logs you provided it looks like the issue is with Firebase. Facebook's deep link url get routed to Firebase's listener instead of Facebook's:
>Failed to process fb479936132915089://authorize#denied_scopes=&granted_scopes=user_friends%2Cemail%2Copenid%2Cpublic_profile&graph_domain=facebook&nonce=AWUXYnXPl1K0wKAn8Ba3L2Tv5tS9ejoPagxmt0ip9ntBoa_Wiwxas9dNfcrUi4DBoS2UTp2GByQfqvGmzzim5ZdzEvg&signed_request=eHtQ-nZvJ9kibGUUxtNllp0Fq3zfABAIE1vYY27nWow.eyJ1c2VyX2lkIjoiMTAxNTcyNTMxMjM2MDMzMzMiLCJjb2RlIjoiQVFCLUYxWXdBaXlmTkRaWXZLZGZ1dTNjaU1EWGhvVG1OcHFKeWdwN3ItYS1UR1NRZjkxbURUb3ZnVTJZVG5vdVZscGtvQ1hTTTNUV3luV1luWFBPVF9jbGpvOVRFcXhCd2c2SEFUTUZzZGo1aldXY2FEcT
Firebase.AppUtil:PollCallbacks()
Firebase.Platform.FirebaseHandler:Update()
